### ISSUE TITLE
[$250] Android - Profile - Domains and Help buttons remain highlighted after navigating back

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#52211]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.59-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Open the Expensify app
2. Tap on "Settings" on the bottom of the screen
3. Tap on "Help"
4. Use device back button to navigate back
5. Verify "Help" button is not highlighted anymore
6. Tap on "Domains" button
7. Navigate back with device back button
8. Verify "Domains" button is not highlighted anymore

## Expected Result:
"Domains" and "Help" button shouldn´t remain selected after navigating back to app

## Actual Result:
After tapping on "Domains" or "Help" buttons and navigating back with device back button, the buttons remain highlighted

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/ef8be3de-8c67-4703-89e0-00c5154d46fb

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021854637211818328663</li>
        <li>Upwork Job ID: 1854637211818328663</li>
        <li>Last Price Increase: 2025-01-09</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @parasharrajat</details>

**Comment by truph01** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
After tapping on "Domains" or "Help" buttons and navigating back with device back button, the buttons remain highlighted

### What is the root cause of that problem?
- When pressing the "Help" button, the icon fills with green to indicate it’s active. This button navigates the user to an external browser, changing appState.isBackground to true. Upon returning to the app, appState.isForeground is set back to true. However, a logic check here uses a cached image instead of loading a new one:

https://github.com/Expensify/App/blob/aedbbde0317b720151a75dbe0b6a7ddea52a20cd/src/components/ImageSVG/index.android.tsx#L21

This results in the icon button remaining highlighted after navigating back.
### What changes do you think we should make in order to solve the problem?
- We can clear the cache data one the appState.isBackground is true in [this](https://github.com/Expensify/App/blob/aedbbde0317b720151a75dbe0b6a7ddea52a20cd/src/components/ImageSVG/index.android.tsx#L16):

```
    const {isBackground} = useAppState();
    useEffect(() => {
        if (isBackground) {
            Image.clearMemoryCache();
        }
    }, [isBackground]);
```

### What alternative solutions did you explore? (Optional)

**Comment by truph01** [comment]
### [Proposal updated](https://github.com/Expensify/App/issues/52211#issuecomment-2463335490)

**Comment by isabelastisser** [comment]
Waiting for more proposals.

**Comment by truph01** [comment]
@parasharrajat What do you think about [my proposal](https://github.com/Expensify/App/issues/52211#issuecomment-2463335490)?

**Comment by truph01** [comment]
I will check it today

**Comment by isabelastisser** [comment]
@parasharrajat, should we close this?

**Comment by truph01** [comment]
> Can you try again

Yes. I can reproduce. I need to investigate the RCA of the issue again

**Comment by isabelastisser** [comment]
@parasharrajat, please provide an  update; thanks!

**Comment by isabelastisser** [comment]
Still waiting for proposals.

**Comment by isabelastisser** [comment]
### FYI, I will be OOO from Dec 23 to Jan 6, so please reassign the issue to another team member for urgency, IF needed.

**Comment by linhvovan29546** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
After tapping on "Domains" or "Help" buttons and navigating back with device back button, the buttons remain highlighted
### What is the root cause of that problem?
The issue may stem from the expo-image implementation in the Android native code. Here's why:
- The tint color updates correctly when the button is pressed.
- The tint color resets correctly when returning from an external browser.
- However, the image does not re-render with the updated `tintColor`.
### What changes do you think we should make in order to solve the problem?
As a workaround:
-  We can add the `tintColor` to the `key` prop of the Image. This will the image to re-render whenever the `tintColor` changes, ensuring the button's visual state updates as expected.
- We should consider adding a flag (`shouldUseKey`) to use the `key` in specific cases where needed. Additionally, consider checking more conditions when navigating back from an external browser, as this issue only occurs when a button is pressed and an external browser is opened
https://github.com/Expensify/App/blob/b23c478cdb918dfe71f867e2c601be6d71478396/src/components/ImageSVG/index.android.tsx#L18-L26
```js
        <Image
            // Caching images to memory since some SVGs are being displayed with delay
            // See issue: https://github.com/Expensify/App/issues/34881
            cachePolicy="memory"
            contentFit={contentFit}
            source={src as ImageSourcePropType}
            style={[{ width, height }, style]}
            key={shouldUseKey && fill ? fill : undefined} // New code
            // eslint-disable-next-line react/jsx-props-no-spreading
            {...tintColorProp}
        />
```

**Note**: _In the POC below (where the solution is applied only to the `Help` icon), notice in the 'After' video at the 00:22 mark that the right icon is highlighted, but the left icon (`Help` icon) is not highlighted._

<details>
 <summary>POC</summary>
<table align="center">
  <tr>
    <td align="center"><strong>Before</strong></td>
    <td align="center"><strong>After</strong></td>
  </tr>
  <tr>
    <td>
      <kbd>
<video src="https://github.com/user-attachments/assets/c322de8d-ea2d-43c0-bc0d-ef9326339fe2" width="100%" controls></video>
      </kbd>
    </td>
    <td>
      <kbd>
<video src="https://github.com/user-attachments/assets/9bd38fc4-bce6-4417-814b-c0eb87393518" width="100%" controls></video>
      </kbd>
    </td>
  </tr>
</table>

</details>

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?
No automated tests are required, as this is a UI-specific bug.

### What alternative solutions did you explore? (Optional)
N/A

**Comment by linhvovan29546** [comment]
> @linhvovan29546 Are you able to reproduce this issue, you before video does not show the issue?

[Updated video](https://github.com/Expensify/App/issues/52211#issuecomment-2571557139), the issue appears at the 00:22

**Comment by isabelastisser** [comment]
@parasharrajat, can you please provide an update? Thanks!

**Comment by linhvovan29546** [comment]
I can't reproduce the issue after upgrading to RN 0.76 (along with upgrading expo-image). It seems this update may have resolved the problem.

**Comment by isabelastisser** [comment]
Closing based on this: https://github.com/Expensify/App/issues/52211#issuecomment-2585005647
